kernel/src/cpu/percpu.rs

```diff
@@ -426,6 +489,19 @@ impl PerCpu {
         Ok(())
     }
 
+    pub fn hv_doorbell(&self) -> Option<&'static HVDoorbell> {
+        unsafe {
+            let hv_doorbell = self.hv_doorbell.get();
+            if hv_doorbell.is_null() {
+                None
+            } else {
+                // The HV doorbell page can only ever be borrowed shared, never
+                // mutable, and can safely have a static lifetime.
+                Some(&*hv_doorbell)
+            }
+        }
```

Comment:
[`as_ref()`](https://doc.rust-lang.org/std/primitive.pointer.html#method.as_ref) already does the null check for us:

```suggestion
        // SAFETY: The HV doorbell page can only ever be borrowed shared, never
        // mutable, and can safely have a static lifetime.
        unsafe {
            self.hv_doorbell.get().as_ref()
        }
```